### ISSUE TITLE
missing changelog ?

### ISSUE THREAD

**Issue by strk** [elpaso/quickwkt#20]
I don't know if this is a QuickWKT bug or a limitation of plugin infrastructure: when looking a tthe "Upgradeable" tab in qgis plugin manager I see QuickWKT installed is 2.2 and available is 2.5, but the following "changelog" is from the 2.2 one so I don't know what I would get by upgrading to 2.5

**Comment by elpaso** [comment]
The changelog seems correct:
https://plugins.qgis.org/plugins/QuickWKT/version/2.5/
and:
https://github.com/elpaso/quickwkt/blob/master/metadata.txt
There is a line for 2.5 too.

**Comment by strk** [comment]
Indeed, so it must be a limitation of the plugin manager to not download the changelog of the proposed updated version, right ?

**Comment by elpaso** [comment]
I checked and the changelog is shown for the installed (current) version. 
It's currently impossible for the installer to access the changelog of the new version before downloading it.

In order to show the changelog in the installer we should change both the installer and the updates XML that at the moment doesn't contain the changelog.

One problem I see is that the updates XML (that is 99% of traffic for our web infrastructure) size would probably double it size.

**Comment by strk** [comment]
Could the ChangeLog of a plugin only be fetched on-demand (if user clicks on a button to now what's new for a given version) ?

Even better if the format of that document could be made formal enough for the backend to only retrive the section between "current" and "new"

I guess this discussion should be moved elsewhere (plugin installer tracker?)

**Comment by elpaso** [comment]
Yes.

It would require some development on both installer and plugin website though.
